### ISSUE TITLE
New config option to not place appMenu items into secondary menu

### ISSUE THREAD

**Issue by alex285** [jderose9/dash-to-panel#83]
On Tweak Tool `Show Application Menu` is set to `off` so apps can draw their fallback menus.
On Dash To Panel the "Show App Menu" either off or on, it display the menu items on appIcon right click menu, in addition to the extra menu (when Show Application Menu is on)

![dash-to-panel-appmenu-bug](https://cloud.githubusercontent.com/assets/11244308/23586175/7bd06c1a-0198-11e7-9990-7068695184b5.png)

basically what I think is a bug here, is that we cannot remove the appMenu items from the appIcon menu. 

btw that is reproduced on GNOME 3.24 beta, saying in case it works differently on 3.22 ..for some reason :)

**Comment by bil-elmoussaoui** [comment]
@alex285 I think what should be correctly done is to make the button insensitive if the show appmenu is enabled on gnome tweak tool with a little tooltip to explain why :)

**Comment by bil-elmoussaoui** [comment]
+ How do you get app menu to work within the app itself? it does not work at all for me.. :(

**Comment by alex285** [comment]
> How do you get app menu to work within the app itself? it does not work at all for me.. :(
i dont know, it just works. It was always working to me. I've seen others have the same issue, no idea why :/

**Comment by jderose9** [comment]
> basically what I think is a bug here, is that we cannot remove the appMenu items from the appIcon menu.

@alex285 This was intentional as I didn't think the appmenu items in the popup menu would bother anyone and need to be removed since they are pretty out of the way, and didn't want to let them ever get into a state where the appmenu items were completely unavailable. Do you see any reason to need to remove them? It does seem that the description for the setting needs to be clarified.

> I think what should be correctly done is to make the button insensitive if the show appmenu is enabled on gnome tweak tool with a little tooltip to explain why :)

@bil-elmoussaoui Unfortunately there's no way for me to tell what the tweak tool setting is enabled or disabled within the extension. The menu gets built every time and there is code on the C side that decides whether to actually display it.

> How do you get app menu to work within the app itself? it does not work at all for me.. :(

@bil-elmoussaoui I'm not sure I understand which behavior you are referring to. If you set the Tweak Tool setting `Top Bar > Show App Menu` to false it will appear at the top left of the application's window. If you right click on a running application, it will now appear in the popup secondary menu as completed for #2 (but this hasn't reviewed/released on gnome.org yet).

**Comment by alex285** [comment]
@jderose9 assuming that we already have set appMenu off from tweak tool

1. it is redundant 
2. it doesn't even make sense (or even work correctly) when the window has not focus
3. some apps (eg gEdit) when you set off the appMenu, they draw different menus anyway, so appMenu on extension doesn't work

my opinion is that you automatically should disable it, if appMenu flag is off, but alternatively you can add an option to have it disabled

**Comment by jderose9** [comment]
@alex285 
When you say it doesn't work when the window isn't focused are you referring to the appMenu item being placed directly on the panel to the left of the other tray items (ie the default behavior of gnome) or the appMenu items being placed in the right-click menu don't work (as I am not aware of a situation where they don't work)?

I can't read the value of the tweak tool setting from the extension so I can't change behavior based on it but if something isn't working properly obviously that needs to be addressed.

I think this is causing enough confusion that the appMenu option under Behavior should just be removed or moved into the Advanced settings with a clearer description, and the items should always appear in the right click menu?

**Comment by alex285** [comment]
@jderose9 
> When you say it doesn't work when the window isn't focused a..

imagine you have a maximized Chrome (focused) window focus, and behind it or in another virtual desktop, 2 GNOME Files windows. Open "About" or "Keyboard" menu and see whats happens. 

But more-over it doesnt even make sense to open "About" when a window has no focus. In Shell, the appMenu is designed to work only for windows that have focus

>I can't read the value of the tweak tool setting from the extension..

not sure if you mean you cant read it, or you dont know how does it, but here's how
https://github.com/GNOME/gnome-tweak-tool/blob/master/gtweak/tweaks/tweak_group_shell.py

> I think this is causing enough confusion that the appMenu.. 

yes it is confusing but if you can read and change the appMenu settings from extension preferences it could make it more clear. But if you cant, my opinion is that it could be a new section, and allow people disable or enable appMenus on right click context menu

irrelevant but this appMenu idea in GNOME 3.0 cause problems for 6 years everywhere :/

**Comment by jderose9** [comment]
@alex285 
> imagine you have a maximized Chrome (focused) window focus, and behind it or in another virtual desktop, 2 GNOME Files windows. Open "About" or "Keyboard" menu and see whats happens.

The about window for Files opens with focus over top of Chrome (which is what I would expect). Are you seeing something different?

**Comment by alex285** [comment]
@jderose9 
> The about window for Files opens with focus over top of Chrome

this is not expected behavior because the "About" is an attached to parent dialog. To reproduce better the weird behavior you can try when Nautilus is on another workspace, then open "About" and then go into overview

**Comment by jderose9** [comment]
> this is not expected behavior because the "About" is an attached to parent dialog. To reproduce better the weird behavior you can try when Nautilus is on another workspace, then open "About" and then go into overview

You are talking about the fact that you are viewing a dialog while on Workspace 2 and it really exists on Workspace 1, correct? So when you open the activities overview, the dialog is not present?  I guess it could be argued that it would be better to go to Workspace 1 first to view the dialog, but the current behavior doesn't seem THAT odd to me and I can't really imagine how frequently people are accessing an About dialog anyway.

The effort to hide them is minimal but the more settings that get added, the more confusing they become and the more has the be spent understanding their nuance. I guess all this appMenu stuff could just get moved to the Advanced popup in the settings so it is out of the way at least.

**Comment by alex285** [comment]
@jderose9 more than this the dialog is supposed to be attached on the main window, but it popups without the app window at all. At least it should bring focus to the main window too

in my opinion the best solution is to be an option show appMenu on appWindows, and when this is enabled it should hide the appMenu items from right click on appIcons

but yeah, advanced settings, or a new appMenu tab, could work fine! i dont think that the options are confusing! come on! we need to chroot every two weeks to fix Linux, you think we'll get confused by that? :)

but to go to the original issue, is that in my use case the extra menu items (that i dont need, since i have the on window appMeny) make the use confusing, because the eye needs to scan more data, till find the option i'm looking for!

**Comment by jderose9** [comment]
@alex285 You can hide the appMenu items from the right click/secondary menu in the settings now. It's in a submenu under the Fine-tune tab. Can you test it out? Thank you!

**Comment by alex285** [comment]
@jderose9 perfect! thank you! i close that!
